### ISSUE TITLE
Can't use phi-2 model with Transformers

### ISSUE THREAD

**Issue by gmonair** [guidance-ai/guidance#544]
**The bug**
I'm trying to load phi-2 with Transformers, it loads the model to GPU but then errors out

**To Reproduce**

```python
from guidance import models, gen
from guidance.library._gen import will_gen
from guidance import capture, one_or_more, any_char, zero_or_more, commit_point, select
import guidance
import re

device = "cuda:0"

phi2 = models.Transformers("microsoft/phi-2", device=device)
```

Error:

```
Loading checkpoint shards: 100%
2/2 [00:31<00:00, 13.67s/it]
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
[<ipython-input-2-f2b5754d2b56>](https://localhost:8080/#) in <cell line: 9>()
      7 device = "cuda:0"
      8 
----> 9 phi2 = models.Transformers("microsoft/phi-2", device=device)

1 frames
[/usr/local/lib/python3.10/dist-packages/guidance/models/transformers/_transformers.py](https://localhost:8080/#) in __init__(self, model, tokenizer, echo, caching, temperature, compute_log_probs, device, **kwargs)
     43             byte_tokens = []
     44             for i in range(len(tkz)):
---> 45                 byte_coded = bytes([tkz.byte_decoder[c] for c in tkz.convert_ids_to_tokens(i)])
     46                 byte_tokens.append(byte_coded)
     47         else:

[/usr/local/lib/python3.10/dist-packages/guidance/models/transformers/_transformers.py](https://localhost:8080/#) in <listcomp>(.0)
     43             byte_tokens = []
     44             for i in range(len(tkz)):
---> 45                 byte_coded = bytes([tkz.byte_decoder[c] for c in tkz.convert_ids_to_tokens(i)])
     46                 byte_tokens.append(byte_coded)
     47         else:

KeyError: ' '
```

**System info (please complete the following information):**
 - OS (e.g. google colab, T4 runtime):
 - Guidance Version (`0.1.8`):

**Comment by marcotcr** [comment]
We'll fix this, but this should work as a workaround in the meantime:
```python
model = 'microsoft/phi-2'
tokenizer = transformers.AutoTokenizer.from_pretrained(model, use_fast=True, trust_remote_code=True)
phi2 = guidance.models.Transformers(model, tokenizer, trust_remote_code=True, device=0)
```

**Comment by gmonair** [comment]
Thanks, I can confirm this works in colab.

**Comment by marcotcr** [comment]
Fixed in 58ec607
